src/main/java/io/neonbee/internal/processor/EntityCollectionProcessor.java

```diff
@@ -284,42 +226,36 @@ public void countEntityCollection(ODataRequest request, ODataResponse response,
-                            } catch (ODataApplicationException | ExpressionVisitException e) {
-                                LOGGER.correlateWith(routingContext).error(e.getMessage(), e);
-                                processPromise.fail(new ODataApplicationException("Exception in filter evaluation",
-                                        HttpStatusCode.INTERNAL_SERVER_ERROR.getStatusCode(), Locale.ENGLISH, e));
-                            }
-                            return false;
-                        }).count();
-                    }).orElse((long) asyncEntity.result().getEntities().size());
-                    /*
-                     * The response body MUST contain the exact count of items matching the request after applying any
-                     * $filter or $search system query options, formatted as a simple primitive integer value with media
-                     * type text/plain. The returned count MUST NOT be affected by $top, $skip, $orderby, or $expand.
-                     * Content negotiation using the Accept request header or the $format system query option is not
-                     * allowed with the path segment /$count.
-                     */
-                    ByteArrayInputStream serializerContent =
-                            new ByteArrayInputStream(String.valueOf(entityCount).getBytes(StandardCharsets.UTF_8));
-                    response.setContent(serializerContent);
-                    response.setHeader(HttpHeader.CONTENT_TYPE, ContentType.TEXT_PLAIN.toContentTypeString());
-                    response.setStatusCode(HttpStatusCode.OK.getStatusCode());
-                    processPromise.complete();
-                });
+
+        fetchEntities(request, edmEntityType, ew -> {
+            try {
+                /*
+                 * The response body MUST contain the exact count of items matching the request after applying any
+                 * $filter or $search system query options, formatted as a simple primitive integer value with media
+                 * type text/plain. The returned count MUST NOT be affected by $top, $skip, $orderby, or $expand.
+                 * Content negotiation using the Accept request header or the $format system query option is not allowed
+                 * with the path segment /$count.
+                 */
+                List<Entity> resultEntityList = applyFilterQueryOption(uriInfo.getFilterOption(), ew.getEntities());
+
+                ByteArrayInputStream serializerContent = new ByteArrayInputStream(
+                        String.valueOf(resultEntityList.size()).getBytes(StandardCharsets.UTF_8));
+                response.setContent(serializerContent);
+                response.setHeader(HttpHeader.CONTENT_TYPE, ContentType.TEXT_PLAIN.toContentTypeString());
+                response.setStatusCode(HttpStatusCode.OK.getStatusCode());
+                getProcessPromise().complete();
```

Comment:
Same here

Reply:
I think it is only called once, because if it passes the try block, the catch block isn't evaluated.